### PR TITLE
Reenabling 9 end to end tests (with fixes)

### PR DESCRIPTION
**Pull request by mathewc**
Supporting fixes here: https://github.com/Azure/azure-webjobs-sdk-extensions/pull/407. I did local testing with that fixed version of the http extension.

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Middleware/HomepageMiddleware.cs [hunk 1]
```diff
@@ -39,7 +40,7 @@ public static bool IsHomepageDisabled
             {
                 IActionResult result = null;
 
-                if (IsHomepageDisabled)
+                if (IsHomepageDisabled || context.Request.IsAntaresInternalRequest())
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/Middleware/HomepageMiddleware.cs]: This was mysteriously removed during the port to core

File: src/WebJobs.Script.WebHost/WebJobsServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -62,6 +63,7 @@ public static IServiceProvider AddWebJobsScriptHost(this IServiceCollection serv
         {
             services.AddWebJobsScriptHostRouting();
             services.AddMvc()
+                .AddJsonOptions(options => options.SerializerSettings.ContractResolver = new DefaultContractResolver())
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/WebJobsServiceCollectionExtensions.cs]: Mvc camel cases JSON. This restores the default. One of the tests I'm enabling was verifying casing.

File: test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SamplesEndToEndTests.cs [hunk 3]
```diff
@@ -174,29 +174,7 @@ public SamplesEndToEndTests(TestFixture fixture)
             Assert.Equal("Testing", TestHelpers.RemoveByteOrderMarkAndWhitespace(result));
         }
 
-        [Fact(Skip = "Needs investigation")]
-        public async Task HttpTrigger_CSharp_Poco_Post_Xml_Succeeds()
```
Comment (mathewc) [inline test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SamplesEndToEndTests.cs]: Decided to just remove the xml goo from the sample. I could have kept it and enabled the test, but I don't think we need it.

File: test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SamplesEndToEndTests.cs [hunk 4]
```diff
@@ -236,7 +214,7 @@ public SamplesEndToEndTests(TestFixture fixture)
             Assert.Equal(HttpStatusCode.OK, response.StatusCode);
         }
 
-        [Fact(Skip = "Needs investigation")]
+        [Fact(Skip = "Node language worker isn't handling duplicate query params properly")]
```
Comment (mathewc) [inline test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SamplesEndToEndTests.cs]: I debugged into this, and even with my fixes for properly deduping query params, node is not doing so. I verified I'm passing the correct query param binding data to the language worker.

I think the language worker is handling the request params itself? @mhoeger ?

File: sample/HttpTrigger-Disabled/function.json [hunk 5]
```diff
@@ -1,5 +1,4 @@
 ﻿{
-    "disabled": true,
```
Comment (mathewc) [inline sample/HttpTrigger-Disabled/function.json]: For the test relying on this, the fixture now sets the requisite AzureWebJobs.HttpTrigger-Disabled.Disabled env var to disable the function

File: test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SamplesEndToEndTests.cs (hunk 3 above)
Comment (brettsam) [reply test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SamplesEndToEndTests.cs]: Is this scenario covered elsewhere? Or do we get it automatically now?

Comment (mathewc) [reply test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SamplesEndToEndTests.cs]: This test was really just testing platform xml handling, not our code. I just added it long ago to verify it worked. However it made the sample grubby with all the DA xml annotations, so removing. Test not really needed.

File: test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SamplesEndToEndTests.cs (hunk 4 above)
Comment (mhoeger) [reply test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SamplesEndToEndTests.cs]: Looking into this and will open a node worker bug if appropriate. Thanks!

**Pull request merged**